### PR TITLE
56 add sign in and sign out flow

### LINKED ISSUE #56: Add sign-in and sign-out flow

### PR DESCRIPTION
**Pull request by pr0xity**
Added the flow and auth providers

### CODE REVIEW

File: project/lib/screens/home_screen.dart [hunk 1]
```diff
@@ -7,12 +7,13 @@ import 'package:project/styles/theme.dart';
 /// Screen/[Scaffold] the user first sees when logging on or finishing signing up.
 /// Displays a bottom tab bar for navigating to other screens.
 class HomeScreen extends StatefulWidget {
+  const HomeScreen({super.key});
+
   /// Named route for this screen.
   static const routeName = "/index";
 
   /// Creates an instance of home screen. A [Scaffold] displaying a
   /// bottom tab bar for other screens.
```
Comment (janitaroyseth) [inline project/lib/screens/home_screen.dart]: You changes position of the constructor, you should move the documentation for it with it 😊

File: project/lib/screens/sign_in_screen.dart [hunk 2]
```diff
@@ -12,11 +13,13 @@ import '../data/example_data.dart';
 
 /// Screen/Scaffold for signing in and signing up .
 class SignInScreen extends StatelessWidget {
+  SignInScreen({super.key});
+
   /// Named route for this screen.
-  static const routeName = "/";
+  static const routeName = "/signin";
 
   /// Creates an instance of [SignInScreen].
```
Comment (janitaroyseth) [inline project/lib/screens/sign_in_screen.dart]: Move documentation to the constructor :)

**Review by janitaroyseth — CHANGES_REQUESTED** [review]
Some small errors for documentation, see comments in files, otherwise great work! Couldn't have done it better myself!

File: project/lib/screens/home_screen.dart (hunk 1 above)
Comment (pr0xity) [reply project/lib/screens/home_screen.dart]: Is fixed in latest commit 👍

File: project/lib/screens/sign_in_screen.dart (hunk 2 above)
Comment (pr0xity) [reply project/lib/screens/sign_in_screen.dart]: Is fixed in latest commit 👍

**Review by janitaroyseth — APPROVED** [review]

**Pull request merged**